src/physical_object.py

```diff
@@ -33,16 +33,39 @@ def get_forces_composition(self):
             force[1] += self.forces[f][1]
         return force
 
+    def collide(self, q):
+        """
+        Dirtily calculates collisions.
+
+        :q: other object
+
+        :returns: new speed
+        """
+        s = self
+        if s is q:
+            return [0., 0.]
+        d = sqrt((s.x - q.x)**2 + (s.y - q.y)**2) * SCALE_FACTOR
+        r1, r2 = s.radius, q.radius
+        if d >= r1 + r2:
+            return [0., 0.]
+        u1, m1, u2, m2 = s.speed, s.mass, q.speed, q.mass
+        ret = [
+            (u1[0] * (m1 - m2) + 2 * m2 * u2[0]) / (m1 + m2),
+            (u1[1] * (m1 - m2) + 2 * m2 * u2[1]) / (m1 + m2)
+        ]
+        return [ret[0] - s.speed[0], ret[1] - s.speed[1]]
+
     def tick(self):
         """Tick the physical object: update its physical properties."""
+        s = self
         # print(str(self))
```

Comment:
The reason to keep it there is to make debugging faster: we have all the debugging functions in correct places, whenever something goes wrong, we just uncomment them and faster navigate to the right place.